pulsar/client_impl_test.go

```diff
@@ -174,6 +174,30 @@ func TestTokenAuth(t *testing.T) {
 	client.Close()
 }
 
+func TestTokenAuthWithSupplier(t *testing.T) {
+	client, err := NewClient(ClientOptions{
+		URL:            serviceURL,
+		Authentication: NewAuthenticationTokenFromSupplier(func() (s string, err error) {
+			token, err := ioutil.ReadFile(tokenFilePath)
+			if err != nil {
+				return "", err
+			} else {
+				return string(token), nil
+			}
```

Comment:
```
if err != nil {
	return "", err
}

return string(token), nil
```

Maybe we can remove `else` branch, right?

Reply:
I don't see much difference between the 2 approaches

Reply:
@merlimat Yes, there is no difference between the 2 approaches. But in `golint`, the following writing is recommended. 

Currently, ci is ok, because we didn't add the check to `.golangci.yml`

Reply:
Got it. Changed